js/framework/AnalyticsConstants.js

```diff
@@ -9,37 +9,356 @@ if (!OO.Analytics.EVENTS)
 {
   const EVENTS =
   {
-    //messages for main video
-    //TODO see if we need to distinguish between video and ad messages
+    /**
+     * @public
+     * @event Analytics#EVENTS#VIDEO_PLAYER_CREATED
+     * @description This message is sent when the player is first created but
+     * may not be ready to play video yet.
+     * @param {Array} paramArray Array of length 1, containing the original parameters
+     * passed into the player.
+     */
     VIDEO_PLAYER_CREATED:           'video_player_created',
+
+    /**
+     * @public
+     * @event Analytics#EVENTS#VIDEO_FIRST_PLAY_REQUESTED
+     * @description This message is sent the first time the user tries to play the video.
+     * In the case of autoplay, it will be sent immediately after the player is ready to play.
+     */
     VIDEO_FIRST_PLAY_REQUESTED:     'video_first_play_requested',
+
+    /**
+     * @public
+     * @event Analytics#EVENTS#VIDEO_PLAY_REQUESTED
+     * @description This message is sent every time there is a request to try and
+     * initiate video playback (except the first time. See VIDEO_FIRST_PLAY_REQUESTED).
+     * This only the request, not when video playback has actually started.
```

Comment:
right, so I was hoping the description of "every time there is a request to play" would be detailed enough.